subprojects/gradle/module-dependencies-graph/src/main/kotlin/com/avito/module/internal/dependencies/ModuleProjectConfigurationDependenciesNode.kt

```diff
@@ -0,0 +1,42 @@
+package com.avito.module.internal.dependencies
+
+import com.avito.module.configurations.ConfigurationType
+import org.gradle.api.Project
+
+internal class ModuleProjectConfigurationDependenciesNode(
+    val project: Project,
+    val configurationType: ConfigurationType,
```

Comment:
Not this model has a narrower scope and limited to a configuration type.
This is to avoid cycles while traversing configurations.

Reply:
I didn't find a test that covers we fix cycle issues

Reply:
Missed it. Will make in another PR.

Reply:
https://github.com/avito-tech/avito-android/pull/873